Add Raspberry Pi Pico WS2812 LED example

WeirdConstructor opened the pull request:
Here is the example for WS2812 LED, like the other example tested with hardware of course, with comments and ASCII diagram :-)

thejpster commented on boards/rp-pico/examples/pico_ws2812_led.rs:
```diff
@@ -0,0 +1,224 @@
+    // Import the `sin` function for a smooth hue animation from the
+    // Pico rp2040 ROM:
+    let sin = unsafe {
+        core::mem::transmute::<_, fn(f32) -> f32>(rp_pico::hal::rom_data::float_funcs::fsin())
```
Why does this need a transmute?

WeirdConstructor replied:
I've really no idea where I got that snippet from. Fixed that, even the unsafe was useless.

thejpster commented on boards/rp-pico/examples/pico_ws2812_led.rs:
```diff
@@ -0,0 +1,224 @@
+    let mut t = 0.0;
+
+    // Bring down the overall brightness of the strip to save
+    // a few amperes (every LED draws ~60mA, RGB means 3 LEDs per
```
Shouldn't this be 'a few milliamps'? An Ampere = An Amp = a enough current to blow the board up! Also applies to the comment at the top of the file.

WeirdConstructor replied:
Yea, my intent with the comments was more about warning the user that they can get quickly into ampere range if they scale up the number of LEDs.

thejpster commented on boards/rp-pico/examples/pico_ws2812_led.rs:
```diff
@@ -0,0 +1,224 @@
+    let v = if v < 0.0 { -v } else { v };
+    let x = c * (1.0 - v);
+    let m = val - c;
+    let (r_, g_, b_) = if (0.0..60.0).contains(&hue) {
```
This might be clearer as `if hue < 60.0 { ... ` (same for each branch).
Also, I wasn't sure why it was `r_` and not just `r`? Underscores make me think of marking variables as unused, although this is at the wrong end for that.

WeirdConstructor replied:
Absolutely, I kinda went from `hue >= 0.0 && hue < 60.0` to something that clippy showed me and did not ask further. Obviously the if cascade demands a much simpler logic X-)

thejpster reviewed (changes requested):
Thank you for this PR! I'm sure many people will be happy to see this example added.

Just a couple of nits I noticed whilst reading though - alternations or clarifiying comments would be fine.

WeirdConstructor commented:
I've cleaned up the hsv2rgb coding style. I've no idea where I got that transmute from, it seems to work fine without! And I also changed the comments to express my intent a bit better.
